src/components/Icon/Icon.js

```diff
@@ -0,0 +1,188 @@
+                                  props.type === "sat_off" ? "-803px -265px" :
+                                  props.type === "exit_white" ? "-75.5px -380px" :
+                                  props.type === "exit_black" ? "-158.5px -380px" :
+                                  props.type === "all_on" ? "-227px -370px" :
+                                  props.type === "sun_on" ? "-310px -370px" :
+                                  props.type === "mon_on" ? "-391px -370px" :
+                                  props.type === "tue_on" ? "-474px -370px" :
+                                  props.type === "wed_on" ? "-557px -370px" :
+                                  props.type === "thu_on" ? "-638px -370px" :
+                                  props.type === "fri_on" ? "-720px -370px" :
+                                  props.type === "sat_on" ? "-803px -370px" :
+                                  props.type === "time_white" ? "-76px -488px" :
+                                  props.type === "time_black" ? "-158px -488px" :
+                                  props.type === "previous" ? "-227px -475px" :
+                                  props.type === "next" ? "-309px -475px" :
+                                  props.type === "minus" ? "-391px -475px" :
+                                  props.type === "plus" ? "-473px -475px" :
+                                  props.type === "add_white" ? "-73px -594px" :
+                                  props.type === "add_black" ? "-158px -594px" :
+                                  props.type === "check_circle" ? "-228px -582px" :
+                                  props.type === "play" ? "-309px -580px" :
+                                  props.type === "cancel_circle" ? "-391px -580px" :
+                                  props.type === "check_border" ? "-474px -580px" :
+                                  props.type === "check_square" ? "-550px -576px" : "0 0"};
+
+`;
+const cx = classNames.bind();
+
+export default function Icon(props) {
+  const { type, alt, style} = props;
+  let size = "xsm"
+  if(type === "drawer" || type === "check_off" || type === "check_on") {
+    size = "sm"
+  }
+  if(type === "all_off" || 
+     type === "sun_off" || 
+     type === "mon_off" || 
+     type === "tue_off" || 
+     type === "wed_off" || 
+     type === "thu_off" || 
+     type === "fri_off" || 
+     type === "sat_off" ||
+     type === "all_on" || 
+     type === "sun_on" || 
+     type === "mon_on" || 
+     type === "tue_on" || 
+     type === "wed_on" || 
+     type === "thu_on" || 
+     type === "fri_on" || 
+     type === "sat_on" ||
+     type === "next" ||
+     type === "minus" ||
+     type === "plus" ||
+     type === "previous" ||
+     type === "check_circle" ||
+     type === "play" ||
+     type === "cancel_circle" ||
+     type === "check_border" )  {
+       size = "md"
+     }
+  if(type === "check_square") {
+    size = "lg"
+  }
```

Comment:
이부분은 useEffect를 써서 컴포넌트가 처음 실행될때 size를 지정하도록 바꾸면 어떨까요? 
```suggestion
let size = 'xsm';
if (
    type === 'all_off'
    || type === 'sun_off'
    || type === 'mon_off'
    || type === 'tue_off'
    || type === 'wed_off'
    || type === 'thu_off'
    || type === 'fri_off'
    || type === 'sat_off'
    || type === 'all_on'
    || type === 'sun_on'
    || type === 'mon_on'
    || type === 'tue_on'
    || type === 'wed_on'
    || type === 'thu_on'
    || type === 'fri_on'
    || type === 'sat_on'
    || type === 'next'
    || type === 'minus'
    || type === 'plus'
    || type === 'previous'
    || type === 'check_circle'
    || type === 'play'
    || type === 'cancel_circle'
    || type === 'check_border'
  ) {
    size = 'md';
  }
  if (type === 'check_square') {
    size = 'lg';
  }
```
또한 lint 에러가 발생하지 않도록 수정하면 위와 같습니다.

Reply:
수정 감사합니다!
좋은 의견 감사합니다. 혹시 저렇게 작성했을 때랑 useEffect사용해서 작성했을때랑 차이점이 있나요?
lifecycle상 렌더링이 된 다음 해당 조건문을 확인하게 돼서 결과의 차이가 있는걸까요?
아니면 lifecycle상 렌더링 될 때 들어가는게 맞는 로직이기 때문에 useEffect에 넣어주는것이 자연스러워?서 그런건가요?
초기화?같은 경우는 쓰는게 당연한데 생각 못하고 넘어가는 경우가 많네요ㅠㅠ

Reply:
`useEffect(() => {}, [type])` 와 같이 작성하였을 때 type이 바뀔 때 마다 실행되는 장점이 있습니다. 

또한 React 함수형 컴포넌트에서 let으로 상태를 정의하면 React에서는 Side effect가 발생할 수 있지 않나요? 사실 이부분에대한 근거를 찾아보려했는데 찾지 못하긴 했습니다..

Reply:
state가 아닌 let으로 상태를 관리시 상태가 업데이트 됐을 때 최신화가 안될 수 도 있는 이슈가 있을거에요!
type을 state로 선언하고 useEffect에서 초기화 해야겠네요

Reply:
네 state 선언하는 부분이랑 useEffect 두개 묶어서 custom hooks로 뺄 수 있을 것 같아요.

Reply:
해당 부분을 재사용 할 것 같진 않은데 custom hooks로 빼는것이 편리할까요?
해당 부분을 재사용성이 있도록 작성하기가 감이 안오네요...ㅠㅠ

Reply:
아 제 생각에는 해당 로직에 해당하는 코드를 간결하게 분리할 수 있다고 생각했는데 custom hooks의 본질적인 필요성인 재사용에 대해서는 좋지 않을 수 있겠네요. 굳이 빼지 않아도 괜찮을 것 같습니다!